pkg/cli/admin/upgrade/status/examples/4.16.0-ec2-control-plane-updated-pdb-prohibits-draining.output

```diff
@@ -25,7 +25,7 @@ build0-gstfj-ci-tests-worker-b-4h7pn      Degraded      Draining   4.16.0-ec.2
 build0-gstfj-ci-tests-worker-b-jv5bg      Degraded      Draining   4.16.0-ec.2   ?      failed to drain node: <node> after 1 hour. Please see machine-config-controller logs for more information
 build0-gstfj-ci-tests-worker-b-kj6gk      Degraded      Draining   4.16.0-ec.2   ?      failed to drain node: <node> after 1 hour. Please see machine-config-controller logs for more information
 build0-gstfj-ci-tests-worker-c-dcz9p      Degraded      Draining   4.16.0-ec.2   ?      failed to drain node: <node> after 1 hour. Please see machine-config-controller logs for more information
-build0-gstfj-ci-prowjobs-worker-d-ddnxd   Unavailable   Pending    ?             ?      node's machine-config daemon state Done is neither Degraded nor Unreconcilable
+build0-gstfj-ci-prowjobs-worker-d-ddnxd   Unavailable   Pending    ?             ?      -
```

Comment:
One more thing we can consider is whether to even show such node above the cut (non-detailed output shows only the "interesting" nodes in the table and leaves out the rest (see the "Omitted additional ..." message on L32).

The "weird" unavailable node does not seem to be actionable, how about hiding it (hide unavailable nodes unless they carry a message). Just a thought, WDYT?

Reply:
> The MCO takes them into account regarding the maxUnavailability.

Good point. Probably  a good argument for keeping it above the cut :+1: 

> If we declare such nodes as Unavailable without a message, it can confuse users, as we don't provide any information. I would rather declare something along the lines "Waiting for the node to be processed by a daemon"/"MCD is processing the node"/"Node is being processed by a Machine Config Daemon"

If something, then I like "Machine Config Daemon is processing the node". It indicates that something is going on, without suggesting that something is wrong, and does not try to describe internal details (like the previous message with states). My only problem is that I think it can be misleading - this situation is something that we never saw in real clusters, we just invented it in test data. But I'm probably good with either an empty message or the short "is processing the node" message.

Reply:
I was thinking too about adding a message like this in the original PR.
The reason is a bit different:
I would like our users (or we) spot the case/bug more easily when a node falls into the situation.
Our conclusion that it is rarely seen but we might be wrong.

Reply:
> The MCO takes them into account regarding the maxUnavailability.

I am not 100% sure, as there may be additional logic for different cases.


The function in the MCO code is used in the `getUnavailableMachines` function: https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/status.go#L461-L486

Which is used in the function `getAllCandidateMachines`: https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/node_controller.go#L1205-L1208
```go
// getAllCandidateMachines returns all possible nodes which can be updated to the target config, along with a maximum
// capacity.  It is the reponsibility of the caller to choose a subset of the nodes given the capacity.
func getAllCandidateMachines(layered bool, config *mcfgv1alpha1.MachineOSConfig, build *mcfgv1alpha1.MachineOSBuild, pool *mcfgv1.MachineConfigPool, nodesInPool []*corev1.Node, maxUnavailable int) ([]*corev1.Node, uint) {
	unavail := getUnavailableMachines(nodesInPool, pool, layered, build)
	if len(unavail) >= maxUnavailable {
		klog.Infof("No nodes available for updates")
		return nil, 0
	}
...
```
---

I have rechecked the logic of `isNodeUnavailable`. I thought we were hitting the `l.ReasonOfUnavailability = ReasonOfUnavailabilityMCDWorkInProgress` line more often, my apologies. With that my opinion is less strong on the matter. It looks like it affects the `maxUnavailability`, so I would welcome a friendly message, and as Petr has mentioned:

> It indicates that something is going on, without suggesting that something is wrong, and does not try to describe internal details (like the previous message with states).

I think I prefer it over an empty message due to the `maxUnavailability`, and the use-case mentioned by Hongkai is reasonable as well.